OpenRA.Mods.Cnc/Traits/Minelayer.cs

```diff
@@ -115,7 +130,7 @@ void IResolveOrder.ResolveOrder(Actor self, Order order)
 				var movement = self.Trait<IPositionable>();
 
 				var minefield = GetMinefieldCells(minefieldStart, cell, Info.MinefieldDepth)
-					.Where(c => movement.CanEnterCell(c, null, BlockedByActor.Immovable) || self.World.FogObscures(c))
+					.Where(c => (movement.CanEnterCell(c, null, BlockedByActor.Immovable) && CanLayMineOnCell(self, c)) || self.World.FogObscures(c))
```

Comment:
The `CanLayMineOnCell` call will usually evaluate much quicker than `CanEnterCell`. Imho we should place `CanLayMineOnCell` here and below first so we can skip `CanEnterCell` if `CanLayMineOnCell` already returned `false`.

Reply:
That was actually fixed by a rewrite I was doing while you were doing the review, sorry :)